rotkehlchen/accounting/history_base_entries.py

```diff
@@ -0,0 +1,261 @@
+        method='spend',
+    )
+    renew_key = str(HistoryEventType.RENEW) + '__' + str(HistoryEventSubType.NONE)
+    result[renew_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        count_entire_amount_spend=True,
+        count_cost_basis_pnl=True,
+        method='spend',
+    )
+    swap_key = str(HistoryEventType.TRADE) + '__' + str(HistoryEventSubType.SPEND)
+    result[swap_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=True,
+        method='spend',
+        accounting_treatment=TxAccountingTreatment.SWAP,
+    )
+    airdrop_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.AIRDROP)
+    result[airdrop_key] = HistoryBaseEntrySettings(
+        taxable=LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    reward_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.REWARD)
+    result[reward_key] = HistoryBaseEntrySettings(
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    return result
+
+
+def history_base_entries_iterator(
+        events_iterator: Iterator[AccountingEventMixin],
+        associated_event: HistoryBaseEntry,
+) -> Iterator[HistoryBaseEntry]:
+    """
+    Takes an iterator of accounting events and transforms it into a history base entries iterator.
+    Takes associated event as an argument to be able to log it in case of errors.
+    """
+    for event in events_iterator:
+        if not isinstance(event, HistoryBaseEntry):
+            log.error(
+                f'At accounting for event {associated_event.notes} with identifier '
+                f'{associated_event.event_identifier} we expected to take an additional '
+                f'event but found a non history base entry event',
+            )
+            return
+
+        yield event
+
+
+class HistoryBaseEntriesAccountant:
+
+    def __init__(
+            self,
+            evm_accounting_aggregators: 'EVMAccountingAggregators',
+            pot: 'AccountingPot',
+    ) -> None:
+        self.evm_accounting_aggregators = evm_accounting_aggregators
+        self.pot = pot
+        self.tx_event_settings: dict[str, HistoryBaseEntrySettings] = {}
+
+    def reset(self) -> None:
+        self.evm_accounting_aggregators.reset()
+        self.tx_event_settings = (  # Using | operator is fine since keys are unique
+            self.evm_accounting_aggregators.get_accounting_settings(self.pot) |  # type: ignore[assignment]  # noqa: E501
+            make_default_accounting_settings(self.pot)
+        )
```

Comment:
As it is now it will work but having this two together opens the door to an error:

1. Developer makes a mistake and the key doesn't have the counterparty for evm events in the settings rules
2. The rule gets imported and it has a callback
3. The callback gets called in a history event

Would it make sense to to split this in a mapping having by key the type or two mappings and have a function that takes the event and checks depending on the type in the correct set of rules? Basically to avoid calling the callback of an evm event in other type of event

Reply:
Hmm yeah, it could make sense indeed